data/json/effects_on_condition/npc_eocs/isherwood_barry_rescue_eocs.json

```diff
@@ -0,0 +1,351 @@
+          {
+            "id": "EOC_BARRY_LEAVE",
+            "effect": [
+              {
+                "u_location_variable": { "global_val": "invasion_map" },
+                "target_params": { "om_terrain": "barry_mi-go_scout_tower_1", "search_range": 500, "z": 0 }
+              },
+              { "u_teleport": { "global_val": "invasion_map" }, "force": true },
+              {
+                "mapgen_update": "remove_barry_and_company",
+                "om_terrain": "barry_mi-go_scout_tower_1",
+                "om_special": "Barry's Mi-Go Scout Tower"
+              }
+            ]
+          }
+        ],
+        "unique_ids": [ "BARRY_MISSION", "CHRIS_MISSION", "LISA_MISSION" ]
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "isherwood_mission_large_damage_retreat",
+    "recurrence": 1,
+    "condition": { "math": [ "currently_invading_map", "==", "1" ] },
+    "effect": [
+      {
+        "if": { "math": [ "isherwood_family_coming", "==", "1" ] },
+        "then": {
+          "u_run_npc_eocs": [
+            {
+              "id": "EOC_ISHERWOODS_TOO_MUCH_DAMAGE_FLEE",
+              "condition": {
+                "or": [
+                  { "math": [ "u_hp('ALL')", "<=", "(u_hp_max('bp_null') * 0.7)" ] },
+                  { "math": [ "u_effect_intensity('bleed')", ">=", "3" ] }
+                ]
+              },
+              "effect": [
+                { "u_teleport": { "global_val": "_teleport_pos" }, "force": true },
+                {
+                  "mapgen_update": "isherwood_rescue_remove_npcs",
+                  "om_terrain": "rural_road",
+                  "om_special": "Barry's Mi-Go Scout Tower"
+                },
+                { "math": [ "currently_invading_map", "=", "0" ] },
+                { "queue_eocs": "isherwood_mission_damage_flee_message", "time_in_future": 1 }
+              ]
+            }
+          ],
+          "unique_ids": [ "CHRIS_MISSION", "LISA_MISSION", "CARLOS_MISSION", "EDDIE_MISSION", "LUKE_MISSION", "JESSE_MISSION" ]
+        }
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "isherwood_mission_player_leaves_map",
+    "global": true,
+    "eoc_type": "EVENT",
+    "required_event": "avatar_moves",
+    "condition": {
+      "and": [
+        { "u_has_mission": "MISSION_ISHERWOOD_CHRIS_1" },
+        {
+          "and": [
+            { "not": { "u_near_om_location": "barry_mi-go_scout_tower_1", "range": 2 } },
+            { "not": { "u_near_om_location": "barry_mi-go_scout_tower_2", "range": 2 } },
+            { "not": { "u_near_om_location": "barry_mi-go_scout_tower_3", "range": 2 } },
+            { "not": { "u_near_om_location": "barry_mi-go_scout_tower_4", "range": 2 } }
+          ]
+        },
+        { "math": [ "currently_invading_map", "==", "1" ] }
+      ]
+    },
+    "effect": [
+      {
+        "u_message": "Are you leaving?  Fine then, we're heading home too!  Meet us back there!",
```

Comment:
I intended for that to be a complaint raised by the Isherwoods if you up and fled, since I assumed they'd not take kindly to you abandoning them near a mi-go tower.